### ISSUE TITLE
Explosive memory consumption under node 0.10.2x

### ISSUE THREAD

**Issue by elmarsto** [twilio/twilio-node#29]
TL;DR; it appears that there is a significant memory leak in this library -- I'm trying to track it down, this is a preliminary report to provide context.

Full version:

I've created a small web app that accepts messages with attachments, stores the attachments securely, and sends a password via twilio to the recipient. The mailer daemon is a single long-running node process that follows a CouchDB _changes feed, looking for new messages.

When I use twilio-node to send the text message, the SMS successfully arrives, _BUT_ the node process' memory consumption balloons to ~700mb resident, ~2Gb all told. 

While not certain, I believe the bug is about 50% replicable. I know for sure it can be reliably reproduced after sending three or four messages. We closed the bug by removing/replacing this library.

Therefore it's off the critical path insofar as my work is concerned. While I can't share the (proprietary) code that's demonstrating the issue, I will attempt to debug twilio-node in my spare time, and will keep you posted. I also have a core file (decanted from the still-running node process via gcore) but again it's got proprietary information in it; I'd need to share it via formal channels with Twilio Inc. (We're a medical services firm, and have to worry about HIPAA privacy legislation in the U.S. and PIPEDA in Canada, and thus have to take security about five times more seriously than I would like -- at least, while I'm debugging ;D)

OS: cloud-provisioned CentOS 6.x, x86_64, 1gb RAM; bug reproducible, even with SELinux disabled. 

Reproduced in: nvm-installed 0.10.21, 0.10.22, 0.10.23, and 0.11.x

twilio-node version: 1.4.0
